week3/tideman.c

```diff
@@ -0,0 +1,217 @@
+    {
+        if (strcmp(candidates[candidate_number], name) == 0)
+        {
+            ranks[rank] = candidate_number;
+
+            return true;
+        }
+    } 
+
+    return false;
+}
+
+void record_preferences(int ranks[])
+{
+    for (int rank = 0; rank < candidate_count; rank++)
+    {
+        for (int rank_after = rank+1; rank_after < candidate_count; rank_after++)
+        {
+            preferences[ranks[rank]][ranks[rank_after]]++;
+        }
+    }
+
+    return;
+}
+
+void add_pairs(void)
+{
+    for (int candidate_number = 0; candidate_number < candidate_count; candidate_number++)
+    {
+        for (int candidate_after = candidate_number+1; candidate_after < candidate_count; candidate_after++)
+        {
+            if (preferences[candidate_number][candidate_after] > preferences[candidate_after][candidate_number])
+            {
+                pairs[pair_count].winner = candidate_number;
+                pairs[pair_count].loser = candidate_after;
+                pair_count++;
+            }
+            else if (preferences[candidate_number][candidate_after] < preferences[candidate_after][candidate_number])
+            {
+                pairs[pair_count].winner = candidate_after;
+                pairs[pair_count].loser = candidate_number;
+                pair_count++;
+            }
+        }
+    }
+
+    return;
+}
+
+void sort_pairs(void)
+{
+    for (int pair_count_minus_one = (pair_count -1); pair_count_minus_one >= 0; pair_count_minus_one--)
+    {
+        for (int pair_number = 0; pair_number <= pair_count_minus_one; pair_number++)
+        {
+            if ((preferences[pairs[pair_number].winner][pairs[pair_number].loser]) < (preferences[pairs[pair_number+1].winner][pairs[pair_number+1].loser]))  
+            {
+                pair temporary = pairs[pair_number];
+                pairs[pair_number] = pairs[pair_number+1];
+                pairs[pair_number+1] = temporary;
+            }
+        }
+    }
+
+    return;
+}
+
+void lock_pairs(void)
+{
+    for (int pair_number = 0; pair_number < pair_count; pair_number++)
+    {
+        if(!has_cycle(pairs[pair_number].loser, pairs[pair_number].winner))
+        {
+            locked[pairs[pair_number].winner][pairs[pair_number].loser] = true;
+        }
+    }
+
+    return;
+}
+
+void print_winner(void)
+{   
+    bool printed = false;
+
+    for(int candidate_number = 0; candidate_number < candidate_count; candidate_number++)
+    {
+        int count = 0;
+
+        for(int locked_winner = 0; locked_winner < candidate_count; locked_winner++)
+        {
+            if(!locked[locked_winner][candidate_number])
+            {
+                count++;
+
+                if(count == candidate_count)
+                {
+                    printf("%s \n",candidates[candidate_number]);
+                    printed = true;
+                    break;
+                }
+            }
+        }
+
+        if (printed) break;
+    }
+
+    return;
```

Comment:
```suggestion
                    return;
                }
            }
        }
    }

    return;
```
Der isPrinted boolean ist hier obsolet, weil du an gegebener Stelle einfach returnen kannst.